File: Mobile/houser/lib/models/Message.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

part 'Message.g.dart';

@JsonSerializable()
class Message extends Equatable{
  Message(this.id, this.senderId, this.matchId, this.sendTime, this.content);

  int id;
  String senderId;
  int matchId;
  DateTime? sendTime;
  String content;

  factory Message.fromJson(Map<String, dynamic> json) => _$MessageFromJson(json);

  Map<String, dynamic> toJson() => _$MessageToJson(this);

  @override
  List<Object?> get props => [id, senderId, matchId, sendTime, content];
}

File: API/HouserAPI/HouserAPI/Models/Message.cs
using System;

namespace HouserAPI.Models
{
    public class Message : Entity
    {
        public User Sender { get; set; }
        public string SenderId { get; set; }
        public Match Match { get; set; }
        public int MatchId { get; set; }
        public DateTime SendTime { get; set; }
        public string Content { get; set; }
    }
}
